src/weights.jl

```diff
@@ -194,6 +194,18 @@ pweights(vs::RealArray) = ProbabilityWeights(vec(vs))
     end
 end
 
+##### Equality tests #####
+
+for w in [AnalyticWeights; FrequencyWeights; ProbabilityWeights; Weights]
```

Comment:
Better use a tuple.

Below, I think it's more standard to do `@eval begin`. Not sure.